docs/applications/hydrogel/langevin_dynamics.py

```diff
@@ -0,0 +1,207 @@
+"""
+Hydrogel Langevin dynamics simulation using matscipy.
+
+This example creates a hydrogel network on a diamond lattice and runs
+molecular dynamics with a Langevin thermostat using ASE.
+
+Physical parameters (in LJ units where kT = 1):
+    - Kuhn length b = 1.0
+    - Chain length N = 50 monomers
+    - Monomer volume v0 = 4*pi/3 (sphere of diameter b)
+    - Flory-Huggins parameter chi = 0.5 (moderate solvent quality)
+    - Coordination number = 4 (diamond lattice)
+
+The crosslinker spacing is chosen as sqrt(N)*b ~ 7.07, which gives
+chains at their equilibrium (Gaussian) extension.
+"""
+
+import numpy as np
+from ase import units
+from ase.build import bulk
+from ase.io import write
+from ase.md.langevin import Langevin
+
+from matscipy.calculators.hydrogel import Hydrogel
+from matscipy.molecules import Molecules
+from matscipy.neighbours import neighbour_list
+
+
+def create_diamond_hydrogel(N=50, b=1.0, n_cells=2):
+    """Create a diamond lattice hydrogel.
+
+    Parameters
+    ----------
+    N : int
+        Number of monomers per chain
+    b : float
+        Kuhn length
+    n_cells : int
+        Number of unit cells in each direction
+
+    Returns
+    -------
+    atoms : ase.Atoms
+        ASE Atoms object with the hydrogel structure
+    molecules : matscipy.molecules.Molecules
+        Bond topology
+    """
+    # Equilibrium end-to-end distance
```

Comment:
This is not the equilibrium end-to-end distance, but it is not a too bad guess that often works to initialize simulations.